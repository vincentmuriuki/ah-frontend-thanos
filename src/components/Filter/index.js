import React from 'react';
import PropTypes from 'prop-types';

const Filter = ({ searchQuery, onChange, onClick }) => {
  const renderInputElement = (id, placeholder) => (
    <input
      type="text"
      name={id}
      id={id}
      placeholder={placeholder}
      className="p-1 m-1"
      onChange={onChange}
    />
  );

  return (
    <div className="container filter-box pt-3 text-center">
      <h6>Search or Filter by:</h6>
      {renderInputElement('title', 'Title', searchQuery)}
      {renderInputElement('tag', 'Tag')}
      {renderInputElement('author', 'Author')}
      <button
        type="button"
        id="btFilter"
        onClick={onClick}
        className="btn btn-primary btn-sm"
      >
        Filter
      </button>
    </div>
  );
};

Filter.propTypes = {
  searchQuery: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Filter;
