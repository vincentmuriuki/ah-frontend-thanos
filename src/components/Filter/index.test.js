import React from 'react';
import { mount } from 'enzyme';
import Filter from '.';

describe('<Filter /> component', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <Filter searchQuery="test" onChange={jest.fn()} onClick={jest.fn()} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
