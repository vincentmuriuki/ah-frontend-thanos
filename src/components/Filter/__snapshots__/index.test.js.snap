// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> component should render correctly 1`] = `
<Filter
  onChange={[MockFunction]}
  onClick={[MockFunction]}
  searchQuery="test"
>
  <div
    className="container filter-box pt-3 text-center"
  >
    <h6>
      Search or Filter by:
    </h6>
    <input
      className="p-1 m-1"
      id="title"
      name="title"
      onChange={[MockFunction]}
      placeholder="Title"
      type="text"
    />
    <input
      className="p-1 m-1"
      id="tag"
      name="tag"
      onChange={[MockFunction]}
      placeholder="Tag"
      type="text"
    />
    <input
      className="p-1 m-1"
      id="author"
      name="author"
      onChange={[MockFunction]}
      placeholder="Author"
      type="text"
    />
    <button
      className="btn btn-primary btn-sm"
      id="btFilter"
      onClick={[MockFunction]}
      type="button"
    >
      Filter
    </button>
  </div>
</Filter>
`;
